File: app/src/main/java/com/kirosc/wifilogger/utilities/JsonUtils.java
package com.kirosc.wifilogger.utilities;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonSyntaxException;
import com.kirosc.wifilogger.ScanResults;

/**
 * Used to convert ScanResults between JSON String.
 */
public class JsonUtils {
    private static GsonBuilder builder = new GsonBuilder();
    private static Gson gson = builder.create();

    /**
     * Convert a ScanResults to a JSON String.
     * @param results The .
     * @return The JSON String.
     */
    public static String toJson(ScanResults results) {
        return gson.toJson(results);
    }

    /**
     * Convert a JSON String to a ScanResults.
     * @param json The JSON String.
     * @return The ScanResults.
     */
    public static ScanResults toScanResults(String json) {
        try {
            return gson.fromJson(json, ScanResults.class);
        } catch (JsonSyntaxException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/kirosc/wifilogger/utils/JsonUtils.kt
package com.kirosc.wifilogger.utils

import com.google.gson.GsonBuilder
import com.google.gson.JsonSyntaxException
import com.kirosc.wifilogger.data.ScanResults



class JsonUtils {
    companion object {
        private val builder = GsonBuilder()
        private val gson = builder.create()


        /**
         * Convert a ScanResults to a JSON String.
         * @param results The ScanResults.
         * @return The JSON String.
         */
        fun toJson(results: ScanResults): String {
            return gson.toJson(results)
        }

        /**
         * Convert a JSON String to a ScanResults.
         * @param json The JSON String.
         * @return The ScanResults.
         */
        fun toScanResults(json: String): ScanResults? {
            return try {
                gson.fromJson(json, ScanResults::class.java)
            } catch (e: JsonSyntaxException) {
                e.printStackTrace()
                null
            }

        }
    }
}